### ISSUE TITLE
A member with the name 'name' already exists on 'OpenAI.Chat.Message'

### ISSUE THREAD

**Issue by MegaMax5000** [RageAgainstThePixel/com.openai.unity#170]
# Bug Report
The following error message occurs when attempting ChatEndpoint.GetCompletionAsync():

JsonSerializationException: A member with the name 'name' already exists on 'OpenAI.Chat.Message'. Use the JsonPropertyAttribute to specify another name.

I apologize if this is not the proper space to bring this up, I could not start a discussion in the Q&A section of the Discussions tab. 

## Overview
Upon looking through the source I see that in Message.cs there is a serialized field "name" as well as using the JsonPropertyAttribute to specify the public "Name" field to serialize as "name".  This might be causing the issue. 

## To Reproduce
The following code reproduces this problem on the "await api.ChatEndpoint.GetCompletionAsync(chatRequest);" line.

```csharp
private async Task<string> SendMessage(string message)
{
    var api = new OpenAIClient(GPT_API_KEY);

    var messages = new List<Message>
    {
        new Message(Role.System, GPT_SYSTEM_ROLE),
        new Message(Role.User, message)
    };
    var chatRequest = new ChatRequest(messages, GPT_MODEL_ID);
    var response = await api.ChatEndpoint.GetCompletionAsync(chatRequest);
    var choice = response.FirstChoice;

    return choice.Message.Content.ToString();
}
```

## Expected behavior
I would expect that error to not occur and for the GetCompletionAsync() to complete as expected.

## Additional context
I am using this package in Unity 2022.3.10f1.

**Comment by MegaMax5000** [comment]
Worth noting when I modified the local package cache version of Message.cs to test a fix, I got the same problem for "role" and "content" and some others once "name" was fixed.

**Comment by StephenHodgson** [comment]
> Worth noting when I modified the local package cache version of Message.cs to test a fix, I got the same problem for "role" and "content" and some others once "name" was fixed.

Yeah this is strange, did you somehow modify the json settings? This should not be possible.

**Comment by MegaMax5000** [comment]
My project is using Newtonsoft Json, could that be causing collisions?

**Comment by StephenHodgson** [comment]
Test installing this in a new project with the same editor version. If it works, then you have another package that is modifying the serialization settings.
